components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt

```diff
@@ -350,4 +356,149 @@ class EventsStorageEngineTest {
+            sendInPings = listOf("store1"),
+            allowedExtraKeys = listOf("foo")
+        )
+
+        // Record the event in the store, without providing optional arguments.
+        EventsStorageEngine.record(
+            metricData = event,
+            monotonicElapsedMs = SystemClock.elapsedRealtime(),
+            extra = mapOf("foo" to "bar")
+        )
+        EventsStorageEngine.testWaitForWrites()
+
+        assertEquals(1, EventsStorageEngine.eventStores["store1"]!!.size)
+        EventsStorageEngine.eventStores.clear()
+
+        // Reinitializing Glean should reload existing events on disk
+        Glean.initialized = false
+        FreshBootDetector.cacheIsFreshBoot = false
+        Glean.initialize(ApplicationProvider.getApplicationContext())
+
+        EventsStorageEngine.record(
+            metricData = event,
+            monotonicElapsedMs = SystemClock.elapsedRealtime()
+        )
+
+        assertEquals(2, EventsStorageEngine.eventStores["store1"]!!.size)
+
+        val snapshot = event.testGetValue()
+        assertEquals(2, snapshot.size)
+        assertEquals("telemetry", snapshot.first().category)
+        assertEquals("test_event", snapshot.first().name)
+        assertTrue(
+            "'extra' keys must be correctly recorded and truncated",
+            mapOf("foo" to "bar") == snapshot.first().extra
+        )
+    }
+
+    @Test
+    fun `flush events on fresh boot`() {
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("OK"))
+
+        resetGlean(getContextWithMockedInfo(), Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port,
+            logPings = true
+        ))
+
+        val event = EventMetricType(
+            disabled = false,
+            category = "telemetry",
+            name = "test_event_clear",
+            lifetime = Lifetime.Ping,
+            sendInPings = listOf("store1"),
+            allowedExtraKeys = listOf("foo")
+        )
+
+        // Record the event in the store, without providing optional arguments.
+        EventsStorageEngine.record(
+            metricData = event,
+            monotonicElapsedMs = SystemClock.elapsedRealtime(),
+            extra = mapOf("foo" to "bar")
+        )
+        EventsStorageEngine.testWaitForWrites()
+
+        // Running Glean.initialize again should trigger the fresh boot detector
+        // and send any pings containing any pending events immediately.
+        FreshBootDetector.cacheIsFreshBoot = true
+        Glean.initialized = false
+        Glean.initialize(
+            getContextWithMockedInfo(),
+            Glean.configuration.copy(
+                serverEndpoint = "http://" + server.hostName + ":" + server.port,
+                logPings = true
+            )
+        )
+
+        // Trigger worker task to upload the pings in the background
+        triggerWorkManager()
+
+        val request = server.takeRequest(20L, TimeUnit.SECONDS)
```

Comment:
Let's make sure that the ping we're receiving is of the expected type and that the expected events are being transported. Given that our metric is declaring just `"store1"`, i'm not sure why this test is passing, honestly :|

Reply:
There's nothing magic about the `events` ping -- event metrics can be on any ping.

But yes, it's a good idea to do some deeper inspection of the ping payload.